src/sensai/data_transformation.py

```diff
@@ -20,15 +21,75 @@ class DataFrameTransformer(ABC):
     (possibly applying the transformation to the original data frame - in-place transformation).
     A data frame transformer may require being fitted using training data.
     """
+    def __init__(self):
```

Comment:
Adding these properties breaks compatibility with persisted instances.
`__setstate__` should be implemented to add these properties and `__getstate__` should be implemented to not store these features, as they represent unnecessary transient state.

Reply:
Done, I think (please have a critical look). I did the same for FeatureGenerator. I don't think the df changes history should be transient, so with the recent change there are not transient attributes.

Reply:
My initial version was wrong but with the last commit it should work (the backwardscompat test runs through). I removed the double underscore since it mangles the attribute name and becomes a bit messy in setstate. Hopefully the single underscore is enough to symbolize that _isFitted should not be accessed directly unless the user knows exactly what they are doing

Reply:
Addressing your question:

>Why don't you think the history should be transient? It's only debugging data, effectively. Why would you want to retain >that when persisting an object? Especially now that you are retaining not only column names but also indices and not >just a pair but an arbitrarily long list (which I don't think is a good idea btw), this could become a lot of redundant data.

I will remove the indices. Concerning the column names change: typically a DFT will be persisted with a model. I want to be able to call the summary methods and to know what will happen in the entire data processing chain. Thus I want to keep the information about how the DFT changes the column names. With the way I implemented the history class, even after deleting the indices we do keep redundant information (columns before and after). But I don't think this can ever be a real downside in practical situations since it will occupy very little ram.